simulariumio/data_objects/display_data.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+
+from ..constants import DISPLAY_TYPE
+from ..exceptions import DataError
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class DisplayData:
+    name: str
+    radius: float
+    display_type: DISPLAY_TYPE
+    url: str
+    color: str
+
+    def __init__(
+        self,
+        name: str,
+        radius: float = None,
+        display_type: DISPLAY_TYPE = DISPLAY_TYPE.NONE,
+        url: str = None,
+        color: str = None,
+    ):
+        """
+        This object contains info about how to display an agent
+
+        Parameters
+        ----------
+        name : str
+            A string display name for this type of agent
+            Default: use names from simulator data if possible
+        radius : float (optional)
+            A float radius for rendering this agent.
+            For fibers, this is the thickness of the line
+            For default agents, this is the scale of the representation
+            Default : 1.0
+        display_type: DISPLAY_TYPE (optional)
+            the type of geometry to display
+            Options: SPHERE, FIBER, PDB, or OBJ
+            Default: If not specified, the Simularium Viewer
+                defaults to SPHERE or FIBER depending on
+                the viz type of each agent
+        url: str (optional)
+            local path or web URL for the geometry file to display,
+            web URLs are required for streaming
+            or loading the trajectory by URL
+            Default: None
+        color: str (optional)
+            the hex value for the color to display, e.g "#FFFFFF"
+            Default: Use default colors from Simularium Viewer
+        """
+        self.name = name
+        # radius defaults to None here so that later,
+        # when it's used to override radius in data,
+        # it's easy to test whether the user has specified it
+        self.radius = radius
+        if not isinstance(display_type, DISPLAY_TYPE):
+            raise DataError(
+                f"In {name}'s DisplayData, display_type is {type(display_type)}"
+                " instead of DISPLAY_TYPE"
+            )
+        self.display_type = display_type
+        self.url = url
+        if color is not None and (
+            (len(color) != 4 and len(color) != 7) or color[0] != "#"
+        ):
+            raise DataError(f"{color} should be provided as '#xxxxxx' or '#xxx'")
+        self.color = color
+
+    @staticmethod
+    def display_type_to_string(display_type: DISPLAY_TYPE) -> str:
```

Comment:
i commented on the commit but this could be a member function of the enum class rather than living here.  alternatively you might be able to give the enum values as strings so as to avoid this function completely.  

but more importantly, a comment would be good just saying that these strings are permanent for file i/o reasons and any change to them must incur a version bump on the format.

Reply:
whoops didn't realize I could use strings as enum values, fixed that and added a comment